contracts/cross-chain/mock/PolygonZkEVMBridgeV2Mock.sol

```diff
@@ -0,0 +1,161 @@
+    }
+
+    function bridgeMessage(
+        uint32 destinationNetwork,
+        address destinationAddress,
+        bool forceUpdateGlobalExitRoot,
+        bytes calldata metadata
+    ) external payable {
+        // If exist a gas token, only allow call this function without value
+        if (msg.value != 0 && address(WETHToken) != address(0)) {
+            revert NoValueInMessagesOnGasTokenNetworks();
+        }
+
+        _bridgeMessage(
+            destinationNetwork,
+            destinationAddress,
+            msg.value,
+            forceUpdateGlobalExitRoot,
+            metadata
+        );
+    }
+
+    function _bridgeMessage(
+        uint32 destinationNetwork,
+        address destinationAddress,
+        uint256 amountEther,
+        bool forceUpdateGlobalExitRoot,
+        bytes calldata metadata
+    ) internal {
+        if (destinationNetwork == networkID) {
+            revert DestinationNetworkInvalid();
+        }
+
+        ++depositCount;
+
+        emit BridgeEvent(
+            _LEAF_TYPE_MESSAGE,
+            networkID,
+            msg.sender,
+            destinationNetwork,
+            destinationAddress,
+            amountEther,
+            metadata,
+            uint32(depositCount)
+        );
+    }
+
+    function claimMessage(
+        bytes32[_DEPOSIT_CONTRACT_TREE_DEPTH] calldata smtProofLocalExitRoot,
+        bytes32[_DEPOSIT_CONTRACT_TREE_DEPTH] calldata smtProofRollupExitRoot,
+        uint256 globalIndex,
+        bytes32 mainnetExitRoot,
+        bytes32 rollupExitRoot,
+        uint32 originNetwork,
+        address originAddress,
+        uint32 destinationNetwork,
+        address destinationAddress,
+        uint256 amount,
+        bytes calldata metadata
+    ) external {
+        // Destination network must be this networkID
+        if (destinationNetwork != networkID) {
+            revert DestinationNetworkInvalid();
+        }
+
+        // Execute message
+        bool success;
+        if (address(WETHToken) == address(0)) {
+            // Native token is ether
+            // Transfer ether
+            /* solhint-disable avoid-low-level-calls */
+            (success, ) = destinationAddress.call{value: amount}(
+                abi.encodeCall(
+                    IBridgeMessageReceiver.onMessageReceived,
+                    (originAddress, originNetwork, metadata)
+                )
+            );
+        } else {
+            // Mint wETH tokens
+            WETHToken.mint(destinationAddress, amount);
+
+            // Execute message
+            /* solhint-disable avoid-low-level-calls */
+            (success, ) = destinationAddress.call(
+                abi.encodeCall(
+                    IBridgeMessageReceiver.onMessageReceived,
+                    (originAddress, originNetwork, metadata)
+                )
+            );
+        }
+
+        if (!success) {
+            revert MessageFailed();
+        }
+
+        emit ClaimEvent(
```

Comment:
Think of a better name for the events here and above. `MessageClaimed` and `MessageBridged` or just `Claimed` and `Bridged` would be better